eunkyung/app.js

```diff
@@ -34,25 +34,141 @@ app.get('/ping', (req, res) => {
 })
 
 app.post('/sign_up', async (req, res, next) => {
-    const { name, password, email } = req.body
+    const { name, password, email, profile_image } = req.body
     await mysqlDataSource.query(
         `INSERT INTO users (
             name, 
             password, 
-            email
+            email,
+            profile_image
             ) 
             VALUES (
+                ?,
                 ?,
                 ?,
                 ?
                 );`,
-        [name, password, email]
+        [name, password, email, profile_image]
     )
     res.status(201).json({ message: 'userCreated' })
 })
 
 
 
+app.post('/posting', async (req, res) => {
+    const { title, content, post_image_url, user_id } = req.body
+    await mysqlDataSource.query(
+        `INSERT INTO posts (
+            title,
+            content,
+            post_image_url,
+            user_id
+        )
+        VALUES (
+            ?,
+            ?,
+            ?,
+            ?
+        );`,
+        [title, content, post_image_url, user_id]
+    )
+    res.status(201).json({ message: 'postCreated' })
+})
+
+app.get('/posts', async (req, res) => {
+    await mysqlDataSource.query(
+        `SELECT
+        users.id as userId,
+        users.profile_image as userprofileImage,
+        posts.user_id as postingId,
+        posts.post_image_url as postingImageUrl,
+        posts.content as postingContent 
+        FROM users INNER JOIN posts ON users.id = posts.user_id`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+
+app.get('/posts/:userId', async (req, res) => {
+    const { userId } = req.params;
+
+    await mysqlDataSource.query(
+        `SELECT
+        users.id as userId,
+        users.profile_image as userprofileImage,
+        pp.postings
+        FROM users
+        LEFT JOIN (
+            SELECT 
+            user_id,
+            JSON_ARRAYAGG(
+                JSON_OBJECT(
+                    'postingId', posts.id,
+                    'postingImageUrl', posts.post_image_url,
+                    'postingContent', posts.content
+                ) 
+            ) 
+        as postings
+        FROM posts 
+        GROUP BY user_id  
+        ) pp ON users.id = pp.user_id
+            WHERE users.id = ${userId};`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+app.patch('/edit/:userId/:postId', async (req, res) => {
```

Comment:
오 Query parameter를 작성하셨군요!👍👍👍

추후에 RESTful API 세션 시간에도 배울 내용이지만, 몇가지 피드백을 드리겠습니다.

- 우선 RESTful API를 작성 (예정)할 땐 엔드포인트 url에 동사를 적지 않습니다. 이미 patch라는 method에 수정을 하겠다는 의미가 내포되어 있기 때문입니다.
따라서 edit 대신 `posts`와 같은 특정 리소스를 적어주면 좋을 것 같습니다.

- userId와 postId도 body 데이터로 받아와도 될 것 같습니다. query-parameter는 특정 리소스를 검색하거나 필터링 하기 위해 클라이언트가 요청할 때 보내는 데이터라고 생각하면 될 것 같습니다.

```suggestion
app.patch('/posts', async (req, res) => {
```